### PR TITLE
fix kube-scheduler typo error

### PR DESCRIPTION
**Pull request by gaorong**
**What this PR does / why we need it**:
 the `Bind` error handling should before the `FinishBinding`  invoked
 
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -367,9 +367,6 @@ func (sched *Scheduler) bind(assumed *v1.Pod, b *v1.Binding) error {
 	// If binding succeeded then PodScheduled condition will be updated in apiserver so that
 	// it's atomic with setting host.
 	err := sched.config.GetBinder(assumed).Bind(b)
```
Comment (neolit123) [inline pkg/scheduler/scheduler.go]: possibly modify it to:
```
if err := sched.config.GetBinder(assumed).Bind(b); err != nil {
...
```

**Comment by neolit123** [comment]
thanks for your PR @GaoRong 

/kind bug
/ok-to-test

**Comment by k82cn** [comment]
Any case for this fix?

File: pkg/scheduler/scheduler.go [hunk 2]
```diff
@@ -384,6 +380,9 @@ func (sched *Scheduler) bind(assumed *v1.Pod, b *v1.Binding) error {
 		})
 		return err
 	}
+	if err := sched.config.SchedulerCache.FinishBinding(assumed); err != nil {
```
Comment (k82cn) [inline pkg/scheduler/scheduler.go]: `FinishBinding` will update field to trigger assumedPod cleanup , this PR will make assumePod here withouth cleanup.

Comment (misterikkit) [reply pkg/scheduler/scheduler.go]: Why is it reasonable to attempt `FinishBinding` if `Bind` returns an err?

**Review by misterikkit — CHANGES_REQUESTED** [review]
The PR description and git commit messages do not really describe the change you are making here. I'd suggest,
- squash the two commits together
- explain why the change is made in the git commit
- put that explanation in the PR description.

Thanks for cleaning up.

File: pkg/scheduler/scheduler.go (hunk 2 above)
Comment (k82cn) [reply pkg/scheduler/scheduler.go]: > Why is it reasonable to attempt FinishBinding if Bind returns an err?

`FinishBinding` set ttl for assumedPod to cleanup.

**Comment by GaoRong** [comment]
I find original intention in #50106

**Comment by k82cn** [comment]
> more comments are expected like this situation

Maybe you can help to add some comments to clarify that to avoid other PRs :)

**Pull request closed without merging**